test/common/multitenancy_load.go

```diff
@@ -0,0 +1,236 @@
+package common
+
+import (
+	goctx "context"
+	"fmt"
+	rhmiv1alpha1 "github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/test/resources"
+	testResources "github.com/integr8ly/integreatly-operator/test/resources"
+	routev1 "github.com/openshift/api/route/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"net/http"
+	"os"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	pollingTime = time.Second * 5
+	realmName   = "rhd"
+)
+
+var multitenantUsers int
+
+func TestMultitenancyLoad(t TestingTB, ctx *TestingContext) {
+	var testUser string
+	var timeOfLogin time.Time
+
+	// get tenant creation time limit
+	tenantCreationTime, err := getTenantCreationTime(t)
+	if err != nil {
+		t.Fatalf("error while getting TENANTS_CREATION_TIMEOUT: %v", err)
+	}
+	// get amount of tenants to be created
+	multitenantUsers, err = getRegisteredTenantsNumber(t)
+	if err != nil {
+		t.Fatalf("error while getting NUMBER_OF_TENANTS")
+	}
+
+	// Get master URL
+	rhmi, err := GetRHMI(ctx.Client, true)
+	if err != nil {
+		t.Fatalf("error getting RHMI CR: %v", err)
+	}
+	masterURL := rhmi.Spec.MasterURL
+
+	// Either setup IDP and create users if in PROW, or just create users if in pipeline
+	err = setupIDP(t, ctx, rhmi)
+	if err != nil {
+		t.Fatal("error while setting up IDP or users %s", err)
+	}
+
+	// Creation of tenants
+	postfix := 1
+	for postfix <= multitenantUsers {
+		isClusterLoggedIn := false
+		routeFound := false
+		isThreeScaleLoggedIn := false
+		var testTimeout time.Duration
+
+		// First user needs more time due to the time IDP takes to be created and available
+		if postfix == 1 {
+			testTimeout = time.Minute * 10
+		} else {
+			testTimeout = tenantCreationTime
+		}
+
+		// Create client for current tenant
+		tenantClient, err := createTenantClient(t, ctx)
+		if err != nil {
+			t.Fatalf("error while creating client for tenant: %v", err)
+		}
+
+		// Build username string eg test-user01, test-user02, test-user100
+		if postfix < 10 {
+			testUser = fmt.Sprintf("%v0%v", DefaultTestUserName, postfix)
+		} else {
+			testUser = fmt.Sprintf("%v%v", DefaultTestUserName, postfix)
+		}
```

Comment:
To keep the padding  consistent  we could replace the if else with  this single line  : 
`testUser = fmt.Sprintf("%v%04v", DefaultTestUserName, postfix)`

Reply:
Great idea thanks @Patryk-Stefanski! Although I think it has to be `testUser = fmt.Sprintf("%v%02v", DefaultTestUserName, postfix)`